tests/test_model.py

```diff
@@ -18,435 +18,439 @@
+            assert model[key] == model_neg[key]
+        else:
+            pytest.fail()
+
+
+def test_CallableNumericalModel():
+    x, y, z = variables('x, y, z')
+    a, b = parameters('a, b')
+
+    model = CallableModel({y: a * x + b})
+    numerical_model = CallableNumericalModel(
+        {y: lambda x, a, b: a * x + b}, [x], [a, b]
+    )
+    assert model.__signature__ == numerical_model.__signature__
+
+    xdata = np.linspace(0, 10)
+    ydata = model(x=xdata, a=5.5, b=15.0).y + np.random.normal(0, 1)
+
+    assert np.array(model(x=xdata, a=5.5, b=15.0)) == pytest.approx(
+        np.array(numerical_model(x=xdata, a=5.5, b=15.0)))
+
+    faulty_model = CallableNumericalModel({y: lambda x, a, b: a * x + b},
+                                          [], [a, b])
+    assert not model.__signature__ == faulty_model.__signature__
+    with pytest.raises(TypeError):
+        # This is an incorrect signature, even though the lambda function is
+        # correct. Should fail.
+        faulty_model(xdata, 5.5, 15.0)
+    
+    # Faulty model whose components do not all accept all of the args
+    faulty_model = CallableNumericalModel(
+        {y: lambda x, a, b: a * x + b, z: lambda x, a: x**a}, [x], [a, b]
+    )
+    assert model.__signature__ == faulty_model.__signature__
+
+    with pytest.raises(TypeError):
+        # Lambda got an unexpected keyword 'b'
+        faulty_model(xdata, 5.5, 15.0)
+
+    # Faulty model with a wrongly named argument
+    faulty_model = CallableNumericalModel(
+        {y: lambda x, a, c=5: a * x + c}, [x], [a, b]
+    )
+    assert model.__signature__ == faulty_model.__signature__
+
+    with pytest.raises(TypeError):
+        # Lambda got an unexpected keyword 'b'
+        faulty_model(xdata, 5.5, 15.0)
+
+    # Correct version of the previous model
+    numerical_model = CallableNumericalModel(
+        {y: lambda x, a, b: a * x + b, z: lambda x, a: x ** a},
+        connectivity_mapping={y: {a, b, x}, z: {x, a}}
+    )
+    # Correct version of the previous model
+    mixed_model = CallableNumericalModel(
+        {y: lambda x, a, b: a * x + b, z: x ** a}, [x],
+        [a, b]
+    )
+   
+    assert np.array(numerical_model(x=xdata, a=5.5, b=15.0)) == pytest.approx(
+        np.array(mixed_model(x=xdata, a=5.5, b=15.0)))
+    zdata = mixed_model(x=xdata, a=5.5, b=15.0).z + np.random.normal(0, 1)
+
+    # Check if the fits are the same
+    fit = Fit(mixed_model, x=xdata, y=ydata, z=zdata)
+    mixed_result = fit.execute()
+    fit = Fit(numerical_model, x=xdata, y=ydata, z=zdata)
+    numerical_result = fit.execute()
+    for param in [a, b]:
+        assert mixed_result.value(param) == pytest.approx(
+            numerical_result.value(param))
+
+        if mixed_result.stdev(param) is  None and numerical_result.stdev(param) is None:
+            assert True
+        elif mixed_result.stdev(param) or  None and numerical_result.stdev(param) is None:
+            assert False
```

Comment:
I assume this is needed because pytes.approx t can't deal with None. If so, this can be replaced by 
```python
if mixed_result.stdev(param) is None or numerical_result.stdev(param) is None:
    assert mixed_result.stdev(param) == numerical_result.stdev(param)
```

Reply:
I implement the check for `None` because I got an error from the pytest. Apparently, there is a case with `None`. But I see now, that the implementation ist wrong :D I changed an `or` with an `is` and a `and` with an `or`:
```python
if mixed_result.stdev(param) is  None and numerical_result.stdev(param) is None:
    assert True
elif mixed_result.stdev(param) is None or numerical_result.stdev(param) is None:
    assert False
else:
    assert mixed_result.stdev(param) == pytest.approx(numerical_result.stdev(param))
```
I would prefer the following implementation, because if only one object is `None` the test has to fail. In your case, you would try to compare the object, if at least one of them is `None`. 
```python
if mixed_result.stdev(param) is not None and numerical_result.stdev(param) is not None:
    assert mixed_result.stdev(param) == pytest.approx(numerical_result.stdev(param))
else
    assert  mixed_result.stdev(param) is None and numerical_result.stdev(param) is None
```
or would you suggest skipping the test if more than one object is `None`?